### PR TITLE
Batch mode

### PR DESCRIPTION
**Pull request by jpatton-USGS**
This PR adds a batch processing mode to neic-locator such that all files with the proper extension in an input directory are located, with the results being written to an output directory. Archiving the input files is optionally supported.

Support for passing an event ID through the locator (to ease the insertion of locator results into a database) was also added.

Support for writing out a csv formatted file in batch mode was added for use in locator evaluation.

### CODE REVIEW

**Review by jmfee-usgs — COMMENTED** [review]
Can locateManyEvents find the many events to be located, and repeatedly call locateSingleEvent (and possibly make locateSingleEvent support csv output...)?  Alternatively, each method should use shared parsing/formatting methods.

File: src/main/java/gov/usgs/locator/LocMain.java [hunk 1]
```diff
@@ -208,106 +391,196 @@ public static void main(String[] args) {
...
+            LOGGER.severe("File: " + fileName);
+            return (1);
+          }
+        } else {
+          // run as LocInput/LocOutput to get access to read/write routines
+          LocInput hydraIn = new LocInput();
+          if (!hydraIn.readHydra(inputPath + File.separatorChar + fileName)) {
+            LOGGER.severe(
+                "Failed to read hydra file: " + inputPath + File.separatorChar + fileName);
+            return (0);
```
Comment (mguy-usgs) [inline src/main/java/gov/usgs/locator/LocMain.java]: Will this keep the errored out file in the current directory, so the app will continue to try reading it and continue getting hung up on it? Below you check for archive to move file or delete if processed, maybe there should be an error dir too? Or did I miss some other handling in the code?

**Review by mguy-usgs — APPROVED** [review]
A couple of comments were made, can be addressed later as appropriate

File: src/main/java/gov/usgs/locator/LocMain.java (hunk 1 above)
Comment (jpatton-USGS) [reply src/main/java/gov/usgs/locator/LocMain.java]: It should be fine given how the function reads files from the directory, but added code to rename bad input files .error to be safe.

**Comment by jpatton-USGS** [comment]
@jmfee-usgs I think I addressed your concerns about locateManyEvents vs locateSingleEvent.

**Comment by jpatton-USGS** [comment]
Fixed what codacy errors I could, it's mostly complaining that functions are long and complicated. It's a locator, of course they are.

**Pull request merged**